src/Streamly/Coreutils/FileTest.hs

```diff
@@ -306,47 +334,97 @@ isSticky = undefined
 --
 -- Like coreutil @test -r file@
 --
--- /Unimplemented/
-isReadable :: FileTest
-isReadable = undefined
-
--- | True if file exists and is writable.  True indicates
--- only that the write flag is on.  The file is not writable on a read-only
--- file system even if this test indicates true.
---
--- Like coreutil @test -w file@
---
--- /Unimplemented/
-isWritable :: FileTest
-isWritable = undefined
-
--- | True if file exists and is executable.  True indicates
--- only that the execute flag is on.  If file is a directory, true
+-- /Pre-release/
+
+isAccessible :: Tuple3 FileMode FileMode FileMode -> FilePath -> IO FileTest
+isAccessible (Tuple3 onr grp oth) path =  do
+    ownerIDMatch <- isOwnerEUID
+    groupIDMatch <- isGroupEGID
+    let ownerPerm = FileTest $ hasMode onr
+        groupPerm = FileTest $ hasMode grp
+        otherPerm = FileTest $ hasMode oth
+    ownerHasPermission <- test path ownerPerm
+    isOwner <- test path ownerIDMatch
+    return $
+        if isOwner && not ownerHasPermission
+        then FileTest (Predicate $ const False)
+        else
+            (groupIDMatch `and` groupPerm)
+            `or` otherPerm
```

Comment:
If `stat` on the file fails then `test` throws an exception. But in case of permission checking we need to say that there is no permission to access the file instead of throwing exception. For that we will have to stat the file explicitly and if a permission-denied errno comes then we need to return file-not-accessible result.

Reply:
It's taken care of in the test function, if a path is not found or does not have permissions it returns False(doesn't throw exceptions).